src/vault/client/http.clj

```diff
@@ -122,40 +123,44 @@
 
 (defn- authenticate-token!
   "Updates the token ref by storing the given auth token."
-  [auth-ref token]
+  [client token]
   (when-not (string? token)
     (throw (IllegalArgumentException. "Token credential must be a string")))
-  (reset! auth-ref {:client-token (str/trim token)}))
+  (reset! (:auth client) {:client-token (str/trim token)}))
 
 
 (defn- authenticate-userpass!
   "Updates the token ref by making a request to authenticate with a username
   and password."
-  [auth-ref api-url credentials]
+  [client credentials]
   (let [{:keys [username password]} credentials]
     (api-auth!
       (str "user " username)
-      auth-ref
-      (do-api-request :post (str api-url "/v1/auth/userpass/login/" username)
-        {:form-params {:password password}
-         :content-type :json
-         :accept :json
-         :as :json}))))
+      (:auth client)
+      (do-api-request :post (str (:api-url client) "/v1/auth/userpass/login/" username)
+        (merge
+          (:http-opts client)
```

Comment:
The second map here is actually the options required to make the request - for example, if the user had set `:form-params` for some reason in `:http-opts` it would break this authentication method. I could see an argument for breaking the content-type params out into an overridable section if the Vault API supports other formats in the future.